evalml/pipelines/time_series.py

```diff
@@ -0,0 +1,82 @@
+import pandas as pd
+
+from evalml.objectives import get_objective
+from evalml.pipelines.regression_pipeline import RegressionPipeline
+from evalml.problem_types import ProblemTypes
+from evalml.utils.gen_utils import any_values_are_nan, drop_nan, pad_with_nans
+
+
+class TimeSeriesRegressionPipeline(RegressionPipeline):
+    problem_type = ProblemTypes.TIME_SERIES_REGRESSION
+
+    def __init__(self, parameters, gap, max_delay, random_state=0):
```

Comment:
I was thinking of doing that once we got to the AutoML integration but it's way better to do this now. I just pushed this change up. The one thing I'd like to hear your thoughts on is whether we should raise an exception when the `pipeline` key is missing. I vote yes because it doesn't make sense to specify default arguments for these pipeline-level arguments.

Reply:
Thanks!

Yeah, the timeseries pipelines need those timeseries parameters in order to function. So yes, if `gap` and `max_delay` aren't included in the `parameters` dict under the `pipeline` key, I think we should raise an exception.

Should we prefix these params with `timeseries`? I.e. `timeseries_gap`, `timeseries_max_delay`. An argument for this would be that we may add other pipeline-level params in the future.

Reply:
Can you explain what you mean? This is how a user would instantiate a TS regression pipeline right now:

```python
    pl = Pipeline({"Delayed Feature Transformer": {"gap": gap, "max_delay": max_delay,
                                                   "delay_features": include_delayed_features,
                                                   "delay_target": include_delayed_features},
                   "pipeline": {"gap": gap, "max_delay": max_delay}})
```

I think renaming to the "pipeline" dict keys to "timeseries_gap" and "timeseries_max_delay" is bit too verbose. We can  add more pipeline level params to the "pipeline" dict as we need without conflicts.

Reply:
Ok, I agree! Let's stick with the example you've provided here.